### PR TITLE
CI: Replace Travis CI with GitHub Actions

### PR DESCRIPTION
**Pull request by DukeNgn**
#### What it does
+ Replace Travis CI with GitHub Actions. Travis is disabled due to a negative credit balance.

image

#### How to test
+ An example build is [here (from fork)](https://github.com/DukeNgn/theia-cpp-extensions/actions/runs/401037042)
+ Deploy step can only be tested after this PR is merged

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: Duc Nguyen <duc.a.nguyen@ericsson.com>

### CODE REVIEW

**Comment by DukeNgn** [comment]
@marcdumais-work With this migration, can you please add `NPM_AUTH_TOKEN` and `GITHUB_TOKEN` to the `secrets` of this repo?

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+name: Build
```
Comment (vince-fugnitto) [inline .github/workflows/build.yml:1]: We should rename the workflow, the `build` is a subset of stages present in the workflow.
The name of the file should also be updated.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        run: |
+          yarn install --no-lockfile
+          yarn prepare
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NODE_OPTIONS: "--max_old_space_size=4096"
+
+      - name: Test
+        shell: bash
+        run: npx lerna run --concurrency=1 --stream --scope "@theia/!(example-)*" test
```
Comment (vince-fugnitto) [inline .github/workflows/build.yml:46]: Why not use the script:

https://github.com/eclipse-theia/theia-cpp-extensions/blob/bb6bb9dfbc36daee0315f6f511c4c4b0fb801eef/package.json#L43-L44

- [upstream changes](https://github.com/eclipse-theia/theia/pull/8780/files#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R65)

Comment (vince-fugnitto) [inline .github/workflows/build.yml:46]: We lost the ability to perform `retries`:

- https://github.com/nick-invision/retry#only-retry-after-error

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+
+      - name: Use Node.js ${{ matrix.node }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node }}
+
+      - name: Before deploy
```
Comment (vince-fugnitto) [inline .github/workflows/build.yml:66]:
```suggestion
      - name: Before Deploy
```

**Review by vince-fugnitto — APPROVED** [review]
The overall changes look good to me 👍

File: .github/workflows/build.yml (hunk 2 above)
Comment (vince-fugnitto) [reply .github/workflows/build.yml]: We cannot use `yarn test:theia` yet until the fix is submitted upstream.
Currently, this approach causes the `macOS` build to fail.

**Comment by marcdumais-work** [comment]
I have added the same `npm publish` token as in the main repo.

**Comment by marcdumais-work** [comment]
Do we know if `NPM_AUTH_TOKEN` is still required? For some reason we did not need to set it in the main repo.

**Comment by DukeNgn** [comment]
> Do we know if `NPM_AUTH_TOKEN` is still required? For some reason we did not need to set it in the main repo.

Can you double-check the `secrets` in the main repo? I see that it's also using `NPM_AUTH_TOKEN` secret [here](https://github.com/eclipse-theia/theia/blob/master/.github/workflows/build.yml#L114), so it should be set there as well.

For this PR, the token is required for the [publish step](https://github.com/eclipse-theia/theia-cpp-extensions/pull/112/files#diff-662dff05c7dfe6681c1007ae601e8573d413a03f9f9d53d951c2cb99caba4fd4R90)

**Review by vince-fugnitto — APPROVED** [review]

**Pull request merged**